bundles/org.openhab.binding.mikrotik/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,431 @@
+				<description>WiFi client MAC address</description>
+			</parameter>
+			<parameter name="ssid" type="text" required="false">
+				<label>SSID</label>
+				<description>
+					Constraining SSID for the WiFi client (optional). If client will connect to another SSID,
+					this thing
+					will stay offline until client reconnects to specified SSID.
+				</description>
+			</parameter>
+			<parameter name="considerContinuous" type="integer" required="false">
+				<label>Consider Home Interval</label>
+				<description>The interval in seconds to treat the client as connected permanently</description>
+				<default>180</default>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+
+	<channel-type id="freeSpace">
+		<item-type>Number</item-type>
+		<label>Free Space</label>
+		<description>Amount of free storage left on device in bytes</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="totalSpace">
+		<item-type>Number</item-type>
+		<label>Total Space</label>
+		<description>Amount of total storage available on device in bytes</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="usedSpace">
+		<item-type>Number</item-type>
+		<label>Used Space %</label>
+		<description>Percentage of used device storage space</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="freeMemory">
+		<item-type>Number</item-type>
+		<label>Free RAM</label>
+		<description>Amount of free memory left on device in bytes</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="totalMemory">
+		<item-type>Number</item-type>
+		<label>Total RAM</label>
+		<description>Amount of total memory available on device in bytes</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="usedMemory">
+		<item-type>Number</item-type>
+		<label>Used RAM %</label>
+		<description>Percentage of used device memory</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="cpuLoad">
+		<item-type>Number</item-type>
+		<label>CPU Load %</label>
+		<description>CPU load percentage</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+
+
+	<channel-type id="interfaceType">
+		<item-type>String</item-type>
+		<label>Type</label>
+		<description>Network interface type</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="interfaceName">
+		<item-type>String</item-type>
+		<label>Interface Name</label>
+		<description>Network interface name</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="comment">
+		<item-type>String</item-type>
+		<label>Comment</label>
+		<description>User-defined comment</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="macAddress">
+		<item-type>String</item-type>
+		<label>MAC Address</label>
+		<description>MAC address of the client or interface</description>
+		<state readOnly="true"/>
+	</channel-type>
```

Comment:
```suggestion
	<channel-type id="macAddress">
		<item-type>String</item-type>
		<label>MAC Address</label>
		<description>MAC address of the client or interface</description>
		<state readOnly="true"/>
		<advanced>true</advanced>
	</channel-type>
```

Since openHAB can auto create items using the "create equipment from thing" feature, the <advanced>true</advanced> will cause it to be hidden until 'show advanced' is ticked. This allows channels that most users will not want to create linked items for when they tick the add all button.

Having loads of channels linked increases the load on the system and also the logging of the data for graphing reasons will add up so consider if any of your channels should be marked as advanced to help users not add them unless they really intend to.

Does that make sense ?

Reply:
Is there a reason to have this as a channel as it's already a parameter?

But the comment about making channels advanced is relevant. Make channels not commenly used advanced. (Note related to the code suggestion: for channel-type the advanced is a attribute, not an element)

Reply:
I almost made the same comment, I believe this is for displaying the mac address of a connected client (something plugged into the router) and not the mac address of the actual thing. I could be wrong.

Reply:
Exposing a `MAC` channel is redundant for WiFi client things as it comes in as a configuration parameter, which means a user knows it for sure.

But for network interface things, which are configured via `name`, this may be useful. Not sure about possible scenario though, but who knows :)

So I'll make it an _advanced_ channel.